### PR TITLE
Update rand(_core)

### PR DESCRIPTION
**Pull request by taiki-e**
This may not be completely fixable. (It seems that some versions have already been yanked...)

### CODE REVIEW

**Comment by seanmonstar** [comment]
I don't really understand why it's recommended to use such version ranges for `rand_core`. Since it defines the traits, a program can't compile if there's multiple `rand_core` versions in the tree, as the traits don't match...

File: futures-util/Cargo.toml [hunk 1]
```diff
@@ -34,8 +34,8 @@ futures-sink-preview = { path = "../futures-sink", version = "=0.3.0-alpha.16",
 futures-select-macro-preview = { path = "../futures-select-macro", version = "=0.3.0-alpha.16", default-features = false, optional = true }
 proc-macro-hack = { version = "0.5", optional = true }
 proc-macro-nested = { version = "0.1.2", optional = true }
-rand = { version = "0.6.4", optional = true }
-rand_core = { version = ">=0.2.2, <0.4", optional = true } # See https://github.com/rust-random/rand/issues/645
+rand = { version = "0.6.5", optional = true }
+rand_core = { version = ">=0.4, <0.5", optional = true } # See https://github.com/rust-random/rand/issues/645
```
Comment (Nemo157) [inline futures-util/Cargo.toml]: This should be just `0.4` now.

**Comment by taiki-e** [comment]
`rand` has a lot of sub crates, which are not always updated at the same time. 
And when releasing a new minor version of `rand_core`, they also publish the old version, which has been changed to re-export the new version. 
So, this may cause compilation to fail if Cargo.toml is not completely up-to-date. This can also be caused by the dependency using an older version.

File: futures-util/Cargo.toml (hunk 1 above)
Comment (taiki-e) [reply futures-util/Cargo.toml]: Yes, `0.4.1` has been yanked.

Comment (Nemo157) [reply futures-util/Cargo.toml]: I just meant that `0.4` is an equivalent bound to `>=0.4, <0.5`.

**Comment by Nemo157** [comment]
https://github.com/rust-random/rand/pull/819 looks like 0.5 may be yanked soon.

File: futures-util/Cargo.toml (hunk 1 above)
Comment (taiki-e) [reply futures-util/Cargo.toml]: Oh, I was thinking that somehow I wrote `">=0.4.1, <0.5"` and opened this PR.

**Comment by taiki-e** [comment]
Hmm... still fail. ICE is will be avoided in #1658. 
#1658 is `out-of-date`, so I rebased it. looks like some `rand*` crates have been yanked, so it may work well without `rand` updates.

**Comment by taiki-e** [comment]
#1658 works well.

**Comment by taiki-e** [comment]
Postpone this PR until there is something upstream.

**Pull request closed without merging**